### ISSUE TITLE
Invalid Product Identifiers?

### ISSUE THREAD

**Issue by jln19** [RevenueCat/purchases-ios#155]
```
DEBUG: Requesting products with identifiers: {(
    "com.<myapp>.MONTH_SUB",
    "com.<myapp>.BASE_SUB"
)}
DEBUG: Products request finished
DEBUG: Valid Products:
DEBUG: Invalid Product Identifiers - (
    "com.<myapp>.BASE_SUB",
    "com.<myapp>.MONTH_SUB"
)
DEBUG: 1 completion handlers waiting on products
INFO: Could not find SKProduct for (
    "com.<myapp>.MONTH_SUB",
    "com.<myapp>.BASE_SUB"
)
INFO: Ensure your products are correctly configured in App Store Connect
```

I have ensured my products are correctly configured in ASC. They are in the "Ready to Submit" state. I can use these same product identifiers to successfully pull products via SwiftyStoreKit and through Apple's API. This leads me to assume that the problem is with my configuration in RevenueCat. But I've gone through the documentation multiple times now over several days and I can't figure out where the problem is. @jeiting I know you diagnose these problems a lot and I've read your blogs. I'm out of things to check and just desperate for some suggestions. Thanks.

**Comment by vegaro** [comment]
Hi @jln19 ! Two things I would suggest are: first, use a real device, and second, double check your bundle id. 

Since this looks more like a configuration issue, can you open a ticket with our support instead?   We will be able to offer you better support that way. 

Thanks!

**Comment by jln19** [comment]
Hi @vegaro , I get the same errors whether on simulator or device. I opened a support ticket about 5 days ago but have not received any response, and have also tried email, so I thought I'd try here. Maybe everyone has been out of town for New Years, so still hoping out hope for some help... 🤞

**Comment by jeiting** [comment]
Hey!

I think I saw your email to me. Sorry, I've been out of the office and
should have forwarded it to support.

I checked Zendesk and couldn't find your ticket but maybe I just missed it.
ᐧ

**Comment by raphaelcruzeiro** [comment]
Just adding my 2 cents here as a new RevenueCat user: I often see this error on a recently installed (or updated) simulator. Quitting the simulator and re-launching makes the product fetching work as expected again.

**Comment by vegaro** [comment]
Yes, the simulators have been really buggy when fetching products, something related to StoreKit and not RevenueCat itself, it's better to test on real devices. Have you been helped via other ways @jln19 ? If that's the case, please close this issue. Thanks!

**Comment by jln19** [comment]
Eventually fixed this by changing my identifiers to `MONTH_SUB` instead of `com.<myapp>.MONTH_SUB`

**Comment by ekurutepe** [comment]
I am seeing the same issue. It happens sporadically like 20-30% of the time. 

@jln19 did changing your product identifiers fix this for you permanently? It's not really a solution for me since this a live app with live products that I'm porting to RevenueCat. So I'd need to create new products etc. etc. I'd love to avoid that.

**Comment by mikezliu** [comment]
I'm seeing this issue on some devices and not others, but seeing it on both simulators and physical devices. Not sure if it's a revenuecat or storekit issue but it's rather hard to debug the cause, just says filed to fetch the products.

**Comment by 0x454447415244** [comment]
The test storekit configuration works fine on the simulator and device. However, physical device is unable to fetch SKProducts on its own. This is happening in one of my projects (others are doing fine).

**Comment by aboedo** [comment]
@mikezliu @0x454447415244 you should probably contact our support team (https://support.revenuecat.com) for this, since it's typically a configuration issue.

**Comment by SebastianKumor** [comment]
I am having this issue on simulator only but but not on device. Any advice ?

**Comment by kasvith** [comment]
I'm also having the same issue in iOS simulator

**Comment by aboedo** [comment]
@SebastianKumor I'm sorry I missed your reply here! I hope you figured it out, but if you haven't, the issue is that iOS simulators are now only compatible with StoreKit Config files. [See this article for more information,](https://support.revenuecat.com/hc/en-us/articles/360041793174-Why-are-products-empty-) but in general terms, you should always try to test with physical devices. 
@kasvith same here. 

If you still want to test in the simulator, you can do so using [StoreKit Config Files as explained here](https://docs.revenuecat.com/docs/apple-app-store#ios-14-only-testing-on-the-simulator). While this is helpful for a lot of things, you'll want to keep in mind that the configuration from the file isn't reflected in App Store Connect, so the best way to make sure that your IAPs are correctly configured is by using a physical device.

**Comment by aboedo** [comment]
I'm locking this issue so that folks who were involved in it back in 2019 don't get pinged for every update in the ticket. If you experience more issues with configuration, please open up a ticket with our [support team here](https://support.revenuecat.com). 

If you experience bugs, please open up a new ticket. 

I hope this helps!
